code/modules/reagents/chemistry/reagents/food_reagents.dm

```diff
@@ -402,6 +402,30 @@
 				M.emote(pick("twitch","giggle"))
 	..()
 
+/datum/reagent/consumable/garlic //NOTE: having garlic in your blood stops vampires from biting you.
+	name = "Garlic Juice"
+	id = "garlic"
+	description = "Crushed garlic. Chefs love it, but it can make you smell bad."
+	color = "#FEFEFE"
+	taste_description = "garlic"
+	metabolization_rate = 0.05 * REAGENTS_METABOLISM
+
+/datum/reagent/consumable/garlic/on_mob_life(mob/living/carbon/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(isvampire(H)) //incapacitating but not lethal. Unfortunately, vampires cannot vomit.
```

Comment:
you'll be able to replace 414-16 with your helper define.

Reply:
I still need a /carbon/human var to check job and use adjust_hygiene(), right?

Reply:
yeah under your helper you'll be able to type as human if true

Reply:
is the current code an ok way of doing that?